distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso

```diff
@@ -102,7 +103,7 @@ type Column
         Invalid_Column_Names.handle_java_exception <| Polyglot_Helpers.handle_polyglot_dataflow_errors <| handle_invalid_value_type <|
             java_column = Java_Problems.with_problem_aggregator Problem_Behavior.Report_Warning java_problem_aggregator->
                 case needs_polyglot_conversion of
-                    True -> Java_Column.fromItems name items expected_storage_type java_problem_aggregator
+                    True -> Java_Column.fromItems name (items.map enso_to_java) expected_storage_type java_problem_aggregator
```

Comment:
I'm a bit worried about performance here. Have you checked `Column_from_vector_*` benchmarks?

Reply:
I've added logic to check the `value_type` and only applying `enso_to_java` if it might be necessary. There's a significant slowdown in the benchmarks in the `value_type=Auto` cases, since it calls `enso_to_java` in those cases.

Before:
```
Benchmarking 'Column_from_vector_1000000.Integers_type_Auto' with configuration: [warmup={2 iterations, 3 seconds each}, measurement={2 iterations, 3 seconds each}]
Measurement avg time:    242.857 ms (+-2.669)
Benchmarking 'Column_from_vector_1000000.Floats_type_Auto' with configuration: [warmup={2 iterations, 3 seconds each}, measurement={2 iterations, 3 seconds each}]
Measurement avg time:    241.672 ms (+-0.935)
```

After:
```
checking value type, but not for auto, auto is worse:
Benchmarking 'Column_from_vector_1000000.Integers_type_Auto' with configuration: [warmup={2 iterations, 3 seconds each}, measurement={2 iterations, 3 seconds each}]
Measurement avg time:    426.575 ms (+-24.659)
Benchmarking 'Column_from_vector_1000000.Floats_type_Auto' with configuration: [warmup={2 iterations, 3 seconds each}, measurement={2 iterations, 3 seconds each}]
Measurement avg time:    351.581 ms (+-2.66)
```

@JaroslavTulach do you have any suggestions for making `enso_to_java` faster? Perhaps implementing it in Java?

Reply:
Can this [comment of mine](https://github.com/enso-org/enso/pull/9950/files#r1605732397) be considered an advice? Does it have any impact on the benchmarks?